File: PlusMinus_HR.js
'use strict';

// process.stdin.resume();
// process.stdin.setEncoding('utf-8');


/*
 * Complete the 'plusMinus' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function plusMinus(arr) {
    
    let plus = 0.0;
    let minus = 0.0;
    let zero = 0.0;

    for(let i of arr){
        if(i<0){
            minus++;
        }
        else if(i>0){
            plus++;
        }else zero++;

    }// End of for each loop

    let av = 0;
    let len = arr.length;

    av = plus/len;
    console.log(av.toFixed(6));
    av=minus/len;
    console.log(av.toFixed(6));
    av = zero/len;
    console.log(av.toFixed(6));

    return;

}

function main() {
    
    let arr = [1,2,0,-1,-2,-3,4,0,9,-5];
    plusMinus(arr);
}


main();

File: PlusMinus_HR.java

import java.util.*;
import java.text.DecimalFormat;
import java.util.ArrayList;

public class PlusMinus_HR {

    /*
     * Complete the 'plusMinus' function below.
     *
     * The function accepts INTEGER_ARRAY arr as parameter.
     */

    public static void plusMinus(List<Integer> arr) {
        
        double plus = 0;
        double minus = 0;
        double zero = 0;
        //Iterate over the array
        for(Integer i: arr){
            if(i.intValue()>0){
                plus++;
            }
            else if(i.intValue()<0){
                minus++;
            }
            else zero++;
        }// End of for-each loop
        
        int len = arr.size();
        if (len == 0){
            System.out.println("0.000000");
            System.out.println("0.000000");
            System.out.println("0.000000");
        } //End of edge condition where list is empty
        
        DecimalFormat df = new DecimalFormat("0.000000");
        
        double av1=0;
        double av2=0;
        double av3=0;
        
        av1 = plus/len;
        System.out.println(plus);
        av2 = minus/len;
        System.out.println(minus);
        av3 = zero/len;
        System.out.println(zero);

        
        System.out.println(df.format(av1));
        System.out.println(df.format(av2));
        System.out.println(df.format(av3));

    }

    public class Solution{

        public static void main(String[] args){
            
            List<Integer> a = new ArrayList<Integer>();

            a.add(1);
            a.add(2);
            a.add(0);
            a.add(-1);
            a.add(-2);
            a.add(7);

            PlusMinus_HR.plusMinus(a);

        }
        
    }

}
